cas-server-core/src/main/java/org/jasig/cas/CentralAuthenticationServiceImpl.java

```diff
@@ -328,7 +316,8 @@ public ServiceTicket grantServiceTicket(
                 this.serviceTicketExpirationPolicy,
                 currentAuthentication != null);
 
-        this.serviceTicketRegistry.addTicket(serviceTicket);
+        this.ticketRegistry.updateTicket(ticketGrantingTicket);

```

Comment:
You are right, I have missed this. The problem was that hazelcast registry which I was testing wasn't returning proxied ticket instance from getTicket method. Fixed this and rolled back changes related to addition of updateTicket:
https://github.com/mduszyk/cas/commit/3496d4fe737ede4a6c90329b07fb5a3a4567879d#diff-8f60343d0dc3c9f1ea4d75ddcfef314fR127